File: res/felippa/people.sc.fsu.edu/~jburkardt/m_src/cube_felippa_rule_test/cube_felippa_rule_test.m
function cube_felippa_rule_test ( )

%*****************************************************************************80
%
%% cube_felippa_rule_test() tests cube_felippa_rule().
%
%  Licensing:
%
%    This code is distributed under the GNU LGPL license. 
%
%  Modified:
%
%    22 December 2018
%
%  Author:
%
%    John Burkardt
%
  addpath ( '../cube_felippa_rule' );

  timestamp ( );
  fprintf ( 1, '\n' );
  fprintf ( 1, 'CUBE_FELIPPA_RULE_TEST()\n' );
  fprintf ( 1, '  MATLAB/Octave version %s.\n', version ( ) );
  fprintf ( 1, '  Test CUBE_FELIPPA_RULE().\n' );

  degree_max = 4;
  cube_monomial_test ( degree_max );

  degree_max = 6;
  cube_quad_test ( degree_max );
%
%  Terminate.
%
  fprintf ( 1, '\n' );
  fprintf ( 1, 'CUBE_FELIPPA_RULE_TEST()\n' );
  fprintf ( 1, '  Normal end of execution.\n' );
  fprintf ( 1, '\n' );
  timestamp ( );

  rmpath ( '../cube_felippa_rule' );

  return
end
function timestamp ( )

%*****************************************************************************80
%
%% timestamp() prints the current YMDHMS date as a timestamp.
%
%  Licensing:
%
%    This code is distributed under the GNU LGPL license.
%
%  Modified:
%
%    14 February 2003
%
%  Author:
%
%    John Burkardt
%
  t = now;
  c = datevec ( t );
  s = datestr ( c, 0 );
  fprintf ( 1, '%s\n', s );

  return
end

File: res/felippa/people.sc.fsu.edu/~jburkardt/f_src/cube_felippa_rule_test/cube_felippa_rule_test.f90
program main

!*****************************************************************************80
!
!! CUBE_FELIPPA_RULE_TEST() tests CUBE_FELIPPA_RULE().
!
!  Licensing:
!
!    This code is distributed under the GNU LGPL license.
!
!  Modified:
!
!    16 July 2009
!
!  Author:
!
!    John Burkardt
!
  implicit none

  integer degree_max

  call timestamp ( )
  write ( *, '(a)' ) ' '
  write ( *, '(a)' ) 'CUBE_FELIPPA_RULE_TEST'
  write ( *, '(a)' ) '  FORTRAN90 version'
  write ( *, '(a)' ) '  Test CUBE_FELIPPA_RULE().'

  degree_max = 4
  call cube_monomial_test ( degree_max )

  degree_max = 6
  call cube_quad_test ( degree_max )
!
!  Terminate.
!
  write ( *, '(a)' ) ' '
  write ( *, '(a)' ) 'CUBE_FELIPPA_RULE_TEST'
  write ( *, '(a)' ) '  Normal end of execution.'
  write ( *, '(a)' ) ' '
  call timestamp ( )

  stop 0
end

